onnxruntime/core/optimizer/conv_bn_fusion.cc

```diff
@@ -131,6 +131,11 @@ Status ConvBNFusion::Apply(Graph& graph, Node& node, RewriteRuleEffect& rule_eff
     graph_utils::AddNodeInput(node, 2, new_conv_B_node_arg);
   }
 
+  // trim off any output defs that are optional in the bn_node before we finalize fusion, as we copy the '
+  // defs across to the Conv node so the output name is maintained. we checked in SatisfyCondition that
+  // none of these optional outputs exist, so it's safe to do this.
+  bn_node.MutableOutputDefs().resize(1);
```

Comment:
>bn_node.MutableOutputDefs().resize(1); [](start = 2, length = 38)

If fusion failed below, what effect this trimming have on the rest of the graph?

Reply:
None of the optional outputs are used, so nothing downstream is depending on them. Also no graceful failure from FinalizeNodeFusion (hence it returns void). Will throw if something is wrong and the loading of the model will fail.